ibm/data_source_ibm_tg_gateway.go

```diff
@@ -0,0 +1,92 @@
+package ibm
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.ibm.com/ibmcloud/networking-go-sdk/transitgatewayapisv1"
+)
+
+func dataSourceIBMTransitGateway() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMTransitGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			tgName: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			tgID: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			tgCrn: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			tgLocation: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			tgCreatedAt: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			tgGlobal: {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			tgStatus: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			tgUpdatedAt: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			tgResourceGroup: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceIBMTransitGatewayRead(d *schema.ResourceData, meta interface{}) error {
+
+	/*
+		id := d.Get(tgName).(string)
+		d.SetId(id)
+		err := resourceIBMTransitGatewayRead(d, meta)
+		if err != nil {
+			return err
+		}
+	*/
+	client, err := transitgatewayClient(meta)
+	if err != nil {
+		return err
+	}
+
+	listTransitGatewaysOptionsModel := &transitgatewayapisv1.ListTransitGatewaysOptions{}
+	listTransitGateways, _, err := client.ListTransitGateways(listTransitGatewaysOptionsModel)
+	if err != nil {
+		return err
+	}
+
+	gwName := d.Get(tgName).(string)
+	for _, tgw := range listTransitGateways.TransitGateways {
+
+		if *tgw.Name == gwName {
+			d.SetId(*tgw.ID)
+			d.Set("crn", tgw.Crn)
+			d.Set("name", tgw.Name)
+			d.Set("location", tgw.Location)
+			d.Set("created_at", tgw.CreatedAt.String())
+
+			if tgw.UpdatedAt != nil {
+				d.Set("updated_at", tgw.UpdatedAt.String())
+			}
+			d.Set("global", tgw.Global)
+			d.Set("status", tgw.Status)
+		}
+	}
```

Comment:
Add  a variable found bool with default false....and check if found false then retrun err could n't find any gateway with specified name

Reply:
added a check for gateway found or not and then reply error message if not found.